### ISSUE TITLE
problem with template type dockerconfigjson

### ISSUE THREAD

**Issue by sp185503** [external-secrets/kubernetes-external-secrets#535]
Hi,

When I define an external secret of type **dockerconfigjson** using GCP Secret Manager the secret isn't populated into Kubernetes correctly.  We're seeing "\" characters and it's preventing the pods from getting pulled from our private registry in Artifactory.

**Example:**

**Secret in Secret Manager:**
```
{"auths": {"mydockerhub.io"{"username":"myusername","password":"mypassword","email":"myemail","auth":"myauth"}}}
```
**External Secret definition:**
```
---
apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:
  name: artifactory
spec:
  backendType: gcpSecretsManager
  projectId: myproject
  template:
    type: kubernetes.io/dockerconfigjson
  data:
    - key: jfrog-image-pull-secret
      name: .dockerconfigjson
```

**K8s secret**

```
 k get secret artifactory -o "jsonpath={.data.\.dockerconfigjson}"  | base64 --decode | jq
{"value": "{\"auths\": {\"mydockerhub.io\"{\"username\":\"myusername\",\"password\":\"mypassword\",\"email\":\"myemail\",\"auth\":\"myauth\"}}}"
}
```

Any idea on what we're doing wrong here?

**Comment by Flydiverny** [comment]
Looks to be wrapped in a `{ "value": "" }` which shouldn't be there? 🤔

**Comment by sp185503** [comment]
@Flydiverny Correct, this is how it looks when I create the secret myself.

```
k get secret artifactory -o "jsonpath={.data.\.dockerconfigjson}"  | base64 --decode | jq
{
  "auths": {
    "mydockerhub.io": {
      "username": "myusername",
      "password": "mypassword",
      "email": "myemail",
      "auth": "myauth"
    }
  }
}
```

**Comment by Flydiverny** [comment]
As a workaround you could probably do 

```yaml
---
apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:
  name: artifactory
spec:
  backendType: gcpSecretsManager
  projectId: myproject
  template:
    type: kubernetes.io/dockerconfigjson
  data:
    - key: jfrog-image-pull-secret
      name: .dockerconfigjson
      property: value
```

Not sure this is a bug in the implementation or something else.

**Comment by sp185503** [comment]
@Flydiverny That worked!  Thank you.

**Comment by MPV** [comment]
For comparison, this works for me:

```yaml
apiVersion: "kubernetes-client.io/v1"
kind: ExternalSecret
metadata:
  name: image-pull-secret-gcr
spec:
  template:
    type: kubernetes.io/dockerconfigjson
  backendType: gcpSecretsManager
  projectId: myproject
  data:
    - key: gcr-pull-secret-for-my-cluster
      name: .dockerconfigjson
```

**Comment by Flydiverny** [comment]
@sp185503 which KES version you using?

**Comment by sp185503** [comment]
@Flydiverny Version 3.3.0

**Comment by Flydiverny** [comment]
Please upgrade your KES version this was previously fixed in version 4.0.0
